spec/lcm/integration/user_provisioning_end_to_end_spec.rb

```diff
@@ -0,0 +1,172 @@
+    @rest_client = LcmConnectionHelper.production_server_connection
+    @domain = @rest_client.domain(LcmConnectionHelper.environment[:prod_organization])
+    @projects = $projects_amount.times.map do |n|
+      opts = {
+        client: @rest_client,
+        title: "LCM User provisioning end-to-end spec project #{@suffix} #{n}",
+        auth_token: LcmConnectionHelper.environment[:prod_token],
+        environment: 'TESTING'
+      }
+      project_helper = Support::ProjectHelper.create(opts)
+      puts 'creating LDM and stuff'
+      project_helper.create_ldm
+      project_helper.load_data
+      puts "created project #{project_helper.project.pid}"
+      project_helper.project
+    end
+    @ub_template_path = File.expand_path('../params/users_brick_e2e.json.erb', __FILE__)
+    @ufb_template_path = File.expand_path('../params/user_filters_brick_e2e.json.erb', __FILE__)
+
+    @domain = @rest_client.domain(LcmConnectionHelper.environment[:prod_organization])
+    @data_product = @domain.data_products('default')
+    @master_project = @rest_client.create_project(title: "Test MASTER project for #{@suffix}", auth_token: LcmConnectionHelper.environment[:prod_token])
+    @segment = @data_product.create_segment(segment_id: "testing-segment-#{@suffix}", master_project: @master_project)
+
+    @user_data = @projects.map do |c|
+      Array.new($user_amount) do |n|
+        {
+          login: "lcm-userprov-e2e-UPPERCASE-#{n}@gooddata.com",
+          project_id: c.pid,
+          position: 'developer'
+        }
+      end
+    end.flatten
+    Support::S3Helper.upload_file(ConfigurationHelper.csv_from_hashes(@user_data), Support::S3Helper::USERS_KEY)
+
+    @test_context = {
+      project_id: @projects.first.pid, # this doesn't really matter for the runtime
+      config: LcmConnectionHelper.environment,
+      s3_bucket: Support::S3Helper::BUCKET_NAME,
+      s3_endpoint: Support::S3Helper::S3_ENDPOINT,
+      s3_key: Support::S3Helper::USER_FILTERS_KEY,
+      sync_mode: 'sync_multiple_projects_based_on_pid',
+      data_product: 'default',
+      users_brick_input: {
+        s3_bucket: Support::S3Helper::BUCKET_NAME,
+        s3_endpoint: Support::S3Helper::S3_ENDPOINT,
+        s3_key: Support::S3Helper::USERS_KEY
+      }
+    }
+  end
+
+  describe '1 - initial MUF setup' do
+    before(:all) do
+      @user_data.map do |u|
+        @domain.add_user(login: u[:login]) unless @domain.users l
+      end
+
+      label, value = uniq_label_with_value @projects.first
+      @mufs = @user_data.map do |u|
+        m = {}
+        m[:login] = u[:login]
+        m[:project_id] = u[:project_id]
+        m[label.title] = value
+        m
+      end.flatten
+
+      Support::S3Helper.upload_file(ConfigurationHelper.csv_from_hashes(@mufs), Support::S3Helper::USER_FILTERS_KEY)
+    end
+
+    it 'setups the MUFs' do
+      BrickRunner.user_filters_brick context: @test_context, template_path: @ufb_template_path
+      @projects.each do |p|
```

Comment:
Identical code found in user_provisioning_end_to_end_spec.rb:101, user_provisioning_end_to_end_spec.rb:148 (mass = 124)